op-challenger/game/fault/solver/rules.go

```diff
@@ -1,38 +1,47 @@
 package solver
 
 import (
+	"context"
 	"errors"
 	"fmt"
+	"math/big"
 
 	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/types"
 )
 
-type actionRule func(game types.Game, action types.Action) error
+type actionRule func(game types.Game, action types.Action, correctTrace types.TraceProvider) error
 
 var rules = []actionRule{
 	parentMustExist,
 	onlyStepAtMaxDepth,
 	onlyMoveBeforeMaxDepth,
 	doNotDuplicateExistingMoves,
 	doNotDefendRootClaim,
+	avoidPoisonedPrestate,
+	detectPoisonedStep,
 }
```

Comment:
Another good rule to add is asserting that there's at most one honest response to a claim. If, for example, the game already contains an honest attack, then the challenger should never contradict this by defending the same parent.

Reply:
Good idea, added as a game validation rule so it can utilities the bidirectional claim tree created when resolving the game to make the check simpler.